publication-commons/src/main/java/no/unit/nva/publication/UserAccessValidationUtil.java

```diff
@@ -0,0 +1,114 @@
+package no.unit.nva.publication;
+
+import static java.util.Objects.nonNull;
+import static no.unit.nva.model.role.Role.CREATOR;
+import static nva.commons.apigateway.AccessRight.EDIT_ALL_NON_DEGREE_RESOURCES;
+import static nva.commons.apigateway.AccessRight.EDIT_OWN_INSTITUTION_RESOURCES;
+import static nva.commons.apigateway.AccessRight.PUBLISH_DEGREE;
+import java.net.URI;
+import java.util.Objects;
+import java.util.Optional;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.PublicationStatus;
+import no.unit.nva.model.Reference;
+import no.unit.nva.model.instancetypes.PublicationInstance;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.pages.Pages;
+import no.unit.nva.publication.model.business.UserInstance;
+import nva.commons.apigateway.AccessRight;
+import nva.commons.apigateway.RequestInfo;
+import nva.commons.apigateway.exceptions.UnauthorizedException;
+
+public class UserAccessValidationUtil {
+
+    public static void validateUserAccessForDeletingPublishedPublication(RequestInfo requestInfo,
+                                                                        Publication publication)
+        throws UnauthorizedException {
+
+        if (!PublicationStatus.PUBLISHED.equals(publication.getStatus())) {
```

Comment:
shouldn't it be draft?

Reply:
No, the idea is that this is a general user validation util for publications of all statuses. If someone uses this method to validate for a publication that is not published, they should be notified. It says in the method name that it is intended for published publications, but that is easy to overlook.